### ISSUE TITLE
Embedding a video breaks the preview with `ERR_ABORTED`

### ISSUE THREAD

**Issue by leonard84** [asciidoctor/asciidoctor-intellij-plugin#640]
## Observed vs. expected behavior

When adding a video via the video tag the preview breaks completely with an error message.

When modifying the document afterwards the preview alternated between working and showing the `ERR_ABORTED` error again.

![grafik](https://user-images.githubusercontent.com/1151511/105229838-cf6fcd00-5b64-11eb-8fed-e7ed5b35d172.png)

![grafik](https://user-images.githubusercontent.com/1151511/105230228-545ae680-5b65-11eb-83da-a735effaaf10.png)

## Steps to reproduce

Add the following video to an asciidoc document.

```asciidoc
video::i5Qu3qYOfsM[youtube]
```

## Environment

Plugin Version: 0.32.7

using the `JCEF Browser`

IntelliJ Details:
IntelliJ IDEA 2020.3.1 (Ultimate Edition)
Build #IU-203.6682.168, built on December 29, 2020
Runtime version: 11.0.9.1+11-b1145.63 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1973M
Cores: 8
Non-Bundled Plugins: com.github.b3er.idea.plugins.arc.browser, org.jetbrains.kotlin, org.asciidoctor.intellij.asciidoc

**Comment by ahus1** [comment]
Sorry to hear you discovered this issue. 

I seems that the non-flicker-refresh for the preview interacts badly with the lazily loading video preview; it seems to load JavaScript/iframes. 

There might be two workarounds for now: 

* skip rendering of video elements using [conditional rendering using ifdef/ifndef](https://intellij-asciidoc-plugin.ahus1.de/docs/users-guide/features/advanced/conditional-rendering.html) to handle the situation where it is previewed in the plugin. 
* disable the option "replace contents without flicker" (see below). Disabling this option will lead to a review flickering on refresh, but it should no longer break

To fix the issue, someone would need to tweak the JavaScript that injects the rendered HTML in the preview. It already applies some kind of double-buffering for MathJax, let's find out how this could be extended to videos.

![image](https://user-images.githubusercontent.com/3957921/105321162-679e9e00-5bc7-11eb-87e2-d4502ba2a1bc.png)

**Comment by leonard84** [comment]
Thanks for the tipps, with regards to the preview of the video, I think it would be fine to completely replace it by a generic placeholder that just says that video is disabled in previews.

**Comment by ahus1** [comment]
I've published release 0.32.14 that solves this issue. 

Once the preview detects an embedded iframe (as this is the case with embedded videos from YouTube and Vimeo) it will not do an in-place refresh. I also improved the performance for this situations where the EDT thread was blocked by some 100-500ms.

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

**Comment by leonard84** [comment]
@ahus1thanks, but would it be possible to have an option to not display any (external) iframe content at all, e.g. by replacing it with a placeholder? Should I create another issue for this?

**Comment by ahus1** [comment]
The current the goal of the preview is to be as close as possible on the real site. This is contrary to a placeholder; users would be surprised to see a placeholder. 

You could create a feature request that asks you to answer the two questions: "Why the new feature should be added" and "How the new feature should work [from a user's perspective]".

There are some technical possibilities, but they depend on the answers on of two questions above. And there is the chance that it will be out of scope or no-one volunteering to implement it.

**Comment by mojavelinux** [comment]
I think some users would be surprised, others would not be surprised. What I think might work is an option that reads "show media previews" or something like that. File managers often have something similar.

**Comment by ahus1** [comment]
It seems that the "ERR_ABORTED" happend also in other situations. Analyzing this for issue #658 I digged into the IntelliJ code in 2020.3.x and found an issue there. Version 0.32.16 should fix this; it restores the pre-2020.3.x behavior. The changes in this issue to disable the in-place replacement when an iframe appears have been reverted as they are no longer necessary.
